### ISSUE TITLE
Fragment loses reactivity when using HDE

### ISSUE THREAD

**Issue by wrnrlr** [ryansolid/dom-expressions#330]
There is a bug in the hyper dom expression runtime where sometimes it loses reactivity after one update.
In my example I use a context, but I am not sure if it can be reproduced in a different way.

[Example](https://codepen.io/wrnrlr/pen/bGPNOKb?editors=1000)

This is strange because as I understand the array is wrapped by `h.Fragment`

**Comment by ryansolid** [comment]
Yeah it is an odd one. It has to do with wrapping and unwrapping of `h` functions which is why we don't see this in other versions.. Like if I change the `render` callto `render(Context(), document.body)` it works but other variations don't. What's happening is the Nested Component is getting prematurely released. I'm pretty sure this is related to the change I made a few versions back to properly handle parent child execution order,.. but there is still a bug here.

**Comment by MarkChrisLevy** [comment]
Hi @ryansolid, any way to fix it or get around that bug?

**Comment by ryansolid** [comment]
We redesigned the h function for the upcoming solid 2.0. I haven't been able to reproduce this issue with it. 1.0 fixing this would require a larger breaking change which puts it outside of scope.
